app/src/main/java/app/revanced/integrations/patches/ReturnYouTubeDislikePatch.java

```diff
@@ -159,20 +161,60 @@ public static CharSequence onLithoTextLoaded(@NonNull Object conversionContext,
     }
 
     /**
-     * Injection point.
+     * Last Shorts dislikes span. Used to temporarily set the shorts span while waiting on fetch call.
+     */
+    @Nullable
+    private static Spanned lastShortsSpan;
+
+    /**
+     * Injection point.  Called when a Shorts dislike is updated.
+     * Handles update asynchronously, otherwise Shorts video will be frozen while the UI thread is blocked.
      *
-     * Called when a Shorts dislike Spanned is created.
+     * @return if RYD is enabled and the TextView was updated
      */
-    public static Spanned onShortsComponentCreated(Spanned original) {
+    public static boolean updateShortsDislikes(@NonNull View view) {
         try {
             if (!SettingsEnum.RYD_ENABLED.getBoolean()) {
-                return original;
+                return false;
             }
-            return ReturnYouTubeDislike.getDislikeSpanForShort(original);
+            LogHelper.printDebug(() -> "updateShortsDislikes");
+
+            TextView textView = (TextView)view;
+            CharSequence text = textView.getText();
+            Spanned textSpan = (text instanceof Spanned)
+                    ? (Spanned) text
+                    : new SpannableString(text);
```

Comment:
The cast might not be necessary if you use TextView as the parameter in case the place where you hook already used check-cast to TextView.